library/Octane/FullReplay.hs

```diff
@@ -0,0 +1,454 @@
+            wasAlreadyAlive = isAlive == Prelude.Just Prelude.True
+            in isOpening && wasAlreadyAlive)
+        -- Remove properties that haven't changed.
+        & Prelude.map (\ replication ->
+            if Parser.replicationState replication == Parser.RSExisting
+            then let
+                actorId = Parser.replicationActorId replication
+                currentState = IntMap.findWithDefault
+                    (Prelude.True, Map.empty) actorId state
+                currentProperties = Prelude.snd currentState
+                newProperties = Parser.replicationProperties replication
+                changes = newProperties
+                    & Map.filterWithKey (\ name newValue -> let
+                        oldValue = Map.lookup name currentProperties
+                        in Prelude.Just newValue /= oldValue)
+                in replication { Parser.replicationProperties = changes }
+            else replication)
+    in frame { Parser.frameReplications = newReplications }
+
+
+reject :: (a -> Prelude.Bool) -> [a] -> [a]
+reject p xs = Prelude.filter (\ x -> Prelude.not (p x)) xs
+
+
+getSpawnedActors :: Parser.Frame -> Aeson.Value
+getSpawnedActors frame = frame
+    & Parser.frameReplications
+    & Prelude.filter (\ replication -> replication
+        & Parser.replicationState
+        & (== Parser.RSOpening))
+    & Prelude.map (\ replication ->
+        ( replication
+            & Parser.replicationActorId
+            & Prelude.show
+            & Text.pack
+        , Aeson.object
+            [ ("Name", replication
+                & Parser.replicationObjectName
+                & Aeson.toJSON)
+            , ("Class", replication
+                & Parser.replicationClassName
+                & Aeson.toJSON)
+            , ("Position", replication
+                & Parser.replicationInitialization
+                & Prelude.fmap Parser.classInitLocation
+                & Monad.join
+                & Aeson.toJSON)
+            , ("Rotation", replication
+                & Parser.replicationInitialization
+                & Prelude.fmap Parser.classInitRotation
+                & Monad.join
+                & Aeson.toJSON)
+            ]
+        ))
+    & Aeson.object
+
+
+getUpdatedActors :: Parser.Frame -> Aeson.Value
+getUpdatedActors frame = frame
+    & Parser.frameReplications
+    & Prelude.filter (\ replication -> replication
+        & Parser.replicationState
+        & (== Parser.RSExisting))
+    & Prelude.map (\ replication ->
+        ( replication
+            & Parser.replicationActorId
+            & Prelude.show
+            & Text.pack
+        , replication
+            & Parser.replicationProperties
+            & Map.map (\ property -> Aeson.object
+                [ ("Type", getPropertyName property)
+                , ("Value", getPropertyValue property)
+                ])
+        ))
+    & Prelude.filter (\ (_, properties) ->
+        Map.null properties)
```

Comment:
Oops! This is the opposite of what I want. That's why the JSON is so small: It doesn't have anything in it!